swagger.yaml

```diff
@@ -146,226 +328,800 @@ paths:
-        example: "DE3BC0B6-D6C4-4E20-917E-95D7EA8C91DC"
-      title:
-        type: string
-        description: "The title of the dataset"
-        example: "CPI"
+      contact:
+        type: array
+        description: "Object containing contact information for this dataset"
+        items:
+          $ref: '#/definitions/Contact'
       description:
         type: string
         description: "A description for a dataset"
-      theme:
-        type: string
-        description: "The release date of this dataset"
+      links:
+        $ref: '#/definitions/DatasetLinks'
       next_release:
         type: string
         description: "The next release date for this dataset"
-      edition:
-        description: "A theme for a dataset"
       periodicity:
         type: string
         description: "The periodicity for a dataset"
       publisher:
-        description: "The publisher of the dataset"
         $ref: '#/definitions/Publisher'
-      latest_url:
+      state:
+        $ref: '#/definitions/State'
+      theme:
         type: string
-        description: "A link to the latest version and edition of the dataset"
-      contact:
-        type: array
-        description: "Object containing contact information for this dataset"
-        items:
-          $ref: '#/definitions/Contact'
-      editions_url:
+        description: "A theme for a dataset"
+      title:
         type: string
-        description: "A URL to all editions for a dataset"
-        example: "http://localhost:8080/datasets/DE3BC0B6-D6C4-4E20-917E-95D7EA8C91DC/editions"
-  Datasets:
+        description: "The title of the dataset"
+        example: "CPI"
+  DatasetLinks:
+    description: "A list of links related to this resource"
+    readOnly: true
     type: object
     properties:
-      items:
-        type: array
-        description: "A list of datasets"
-        items:
-          $ref: '#/definitions/Dataset'
-  Edition:
+      editions:
+        description: "A URL to all editions for this dataset"
+        example: "http://localhost:8080/datasets/DE3BC0B6-D6C4-4E20-917E-95D7EA8C91DC/editions"
+        type: string
+      latest_version:
+        $ref: '#/definitions/LatestVersionLink'
+      self:
+        $ref: '#/definitions/SelfLink'
+  LatestVersionLink:
     type: object
+    readOnly: true
     properties:
       id:
+        description: "The unique id for the latest version of a dataset"
+        type: "string"
+      link:
+        description: "A link to the latest version and edition of the dataset"
+        example: "http://localhost:8080/datasets/DE3BC0B6-D6C4-4E20-917E-95D7EA8C91DC/editions/2017/versions/2"
         type: string
-        description: "An unique id for a dataset edtion"
+  Edition:
+    type: object
+    properties:
+      edition:
+        description: "The edition of the dataset"
         example: "2017"
-      versions_url:
+        readOnly: true
+        type: string
+      id:
```

Comment:
From the API perspective, the id at this level is the same as the edition, probably dont need the edition

Reply:
No id is a uuid

Reply:
After discussion with @eldeal we will discuss this further and create future work to tidy up id's (whether they get returned, or an edition_id equal to the edition value gets returned etc.)